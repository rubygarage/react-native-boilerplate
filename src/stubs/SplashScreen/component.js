import React from 'react';
import styled from 'styled-components/native';
import colors from '../../assets/theme/colors';

const SplashScreen = () => (
  <SplashScreen.Container>
    <SplashScreen.LogoText>GO</SplashScreen.LogoText>
  </SplashScreen.Container>
);

SplashScreen.Container = styled.View`
  flex-grow: 1;
  background-color: ${colors.white};
  justify-content: center;
  align-items: center;
`;

SplashScreen.LogoText = styled.Text`
  font-size: 72px;
  line-height: 96px;
  color: ${colors.black};
  font-weight: bold;
`;

SplashScreen.navigationOptions = () => ({
  headerTitle: 'SplashScreen',
});

export default SplashScreen;
