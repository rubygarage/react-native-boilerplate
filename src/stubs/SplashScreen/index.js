import SplashScreen from './component';

export default SplashScreen;
