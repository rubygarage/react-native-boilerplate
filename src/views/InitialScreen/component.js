import { useEffect } from 'react';
import PropTypes from 'prop-types';
import SplashScreen from 'react-native-splash-screen';

const InitialScreen = ({ navigation }) => {
  useEffect(() => {
    SplashScreen.hide();
    navigation.navigate('Home');

    // eslint-disable-next-line
  }, []);

  return null;
};

InitialScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};

export default InitialScreen;
