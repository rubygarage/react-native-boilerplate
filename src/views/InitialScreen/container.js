import React from 'react';
import PropTypes from 'prop-types';
import SplashScreen from 'react-native-splash-screen';

class InitialScreen extends React.Component {
  componentDidMount() {
    const { navigation: { navigate } } = this.props;
    SplashScreen.hide();
    navigate('Home');
  }

  render() {
    return null;
  }
}

InitialScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};

export default InitialScreen;
