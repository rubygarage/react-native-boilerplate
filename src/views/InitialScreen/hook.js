import { useEffect } from 'react';
import SplashScreen from 'react-native-splash-screen';
import { useNavigation } from '@react-navigation/native';

const useContainer = () => {
  const navigation = useNavigation();

  /**
   * Mounting
   */
  const didMount = () => {
    SplashScreen.hide();
    navigation.navigate('Home');
  };

  useEffect(didMount, []);

  return {
    navigation,
    didMount,
  };
};

export default useContainer;
