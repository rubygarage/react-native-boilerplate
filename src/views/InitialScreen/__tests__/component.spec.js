import React from 'react';
import { shallow } from 'enzyme';
import SplashScreen from 'react-native-splash-screen';

import InitialScreen from '../component';

jest.mock('react-native-splash-screen', () => ({
  hide: jest.fn(),
}));

describe('InitialScreen component', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
  };

  let component = null;

  beforeEach(() => {
    component = shallow(<InitialScreen {...props} />);
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('useEffect mounting', () => {
    expect(SplashScreen.hide).toHaveBeenCalled();
    expect(props.navigation.navigate).toHaveBeenCalledWith('Home');
  });
});
