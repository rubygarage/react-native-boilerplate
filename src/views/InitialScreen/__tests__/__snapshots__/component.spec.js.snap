// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialScreen component renders correctly 1`] = `""`;
